raiden/network/proxies/secret_registry.py

```diff
@@ -82,10 +82,17 @@ def register_secret_batch(self, secrets: List[typing.Secret]):
             log.debug('registerSecretBatch skipped', **log_details)
             return
 
-        log.debug('registerSecretBatch called', **log_details)
+        gas_limit = self.proxy.estimate_gas('registerSecretBatch', secrets)
```

Comment:
Bit unfortunate that we call `estimate_gas` twice after this change, do you think we can raise a new exception in `safe_gas_limit` when a parameter is `None` to avoid doing that?

Reply:
Why do we call it twice? I don't see it.
`safe_gas_limit` does not allow a `None` parameter right now. It has an [assertion](https://github.com/raiden-network/raiden/pull/2988/files#diff-105d8fd25b5024d276fe9daa496c628eR270) for that.

Reply:
Nvm, my bad